h2o-core/src/main/java/water/fvec/Frame.java

```diff
@@ -1701,18 +1725,11 @@ private void appendColumnName(StringBuilder sb, String name) {
         final String[] originalDomain = vec.domain();
         final String[] escapedDomain = new String[originalDomain.length];
 
-        boolean escapingRequired = false;
         for (int level = 0; level < originalDomain.length; level++) {
-          escapedDomain[level] = escapeQuotesForCsv(originalDomain[level]);
-          escapingRequired = escapingRequired || !escapedDomain[level].equals(originalDomain[level]);
+          escapedDomain[level] = '"' + escapeQuotesForCsv(originalDomain[level]) + '"';
```

Comment:
Nice fix.

Possible minor speedup: the `escapeQuotesForCsv` method contains the following check: `if (!_parms._escapeQuotes) return unescapedString;`

If that check is done before the method is called, allocation of new stackframe might be saved. Not sure about how JVM optimizes that scenario thogh. Just an idea.